WordPress/gutenberg
# Navigation: Fix 'ariaLabel' block support

## What?
Added text control field to navigation block to specify other accessible name apart from menu name.

## Why?
Closes https://github.com/WordPress/gutenberg/issues/66935

## How?
- created `accessibleMenuName` attribute and storing name provided in text field.
- setting `aria-lable` to with provided attribute and if that's empty then applying fallback label with menu name.

## Testing Instructions
1. open post editor & add navigation block
2. go to `Settings > Advance` and add any text to `Accessible menu name` field.
3. check front end navigation block aria-label will be updated with new provided text.


## Screenshots or screencast 

https://github.com/user-attachments/assets/8f20b10b-9dba-48d2-8242-d1a99038771e
